openedx/features/course_duration_limits/models.py

```diff
@@ -129,27 +126,18 @@ def enabled_for_enrollment(cls, enrollment=None, user=None, course_key=None):
             elif has_staff_roles(user, course_key):
                 return False
 
+        no_masquerade = get_course_masquerade(user, course_key) is None
+        student_masquerade = is_masquerading_as_specific_student(user, course_key)
```

Comment:
Assuming these will do the right thing when there is no enrollment?

Reply:
yes these give expected results when enrollment is None